sbomgrader/translate/translation_map.py

```diff
@@ -232,26 +314,38 @@ def convert(
             )
             for fallback in doc.sbom_format_fallback
         ), f"This map cannot convert from {doc.sbom_format}."
+        # Preprocess
+        dict_ = doc.doc
+        for preprocessing_func in self.preprocessing_funcs.get(
+            self._input_format(doc), []
+        ):
+            res = preprocessing_func(doc.doc)
+            if res:
+                dict_ = res
+        doc = Document(dict_)
```

Comment:
Please help me to understand this because I am not sure what is a goal of this. There is a for loop that iterates over a preprocessing function. A document is given as an input to the fucntion. If a results exists it is stored into `dict_` variable. And at the end a new `Document` is created based on the output.

Does it mean that only the last preprocessing fucntion is used to generate the document? Because the loop overwrites the previous iterations results.

Reply:
Hmm, you are right, this is not correct. While currently all of the functions get correctly applied, it's simply because I used methods that mutate the document in-place. I will have to fix this bit. Thank you for spotting this.